+nosnoc/+discrete_time_problem/Cls.m

```diff
@@ -0,0 +1,1600 @@
+                    v_prev = x_prev(dims.n_q + 1:end);
+                    x_lbp = obj.w.x(ii,jj,0);
+                    q_lbp = x_lbp(1:dims.n_q);
+                    v_lbp = x_lbp(dims.n_q + 1:end);
+                    obj.g.q_continuity(ii,jj) = {q_prev - q_lbp};
+                    z_impulse_ij = obj.z_impulse(ii,jj);
+
+                    if (jj ~= 1 || ~opts.no_initial_impacts)
+                        obj.g.impulse(ii,jj) = {dcs.g_impulse_fun(q_lbp,v_lbp,v_prev,z_impulse_ij, v_global, p)};
+                    else
+                        obj.g.v_continuity(ii,jj) = {v_lbp-v_prev};
+                    end
+                    % additional y_eps constraint
+                    if opts.eps_cls > 0
+                        x_eps = vertcat(q_lbp + h * opts.eps_cls * v_lbp, v_lbp);
+                        obj.g.f_c_eps(ii,jj) = {dcs.f_c_fun(x_eps), 0, inf};
+                    end
+                    
+                    switch opts.rk_representation
+                      case RKRepresentation.integral
+                        % In integral representation stage variables are states.
+                        x_ij_end = opts.D_rk(1)*x_lbp;
+                        for kk=1:opts.n_s
+                            x_ijk = obj.w.x(ii,jj,kk);
+                            z_ijk = obj.w.z(ii,jj,kk);
+                            z_alg_ijk = obj.z_alg(ii,jj,kk);
+                            z_alg_f_x_ijk = obj.z_alg_f_x(ii,jj,kk);
+
+                            fj = s_sot*dcs.f_x_fun(x_ijk, z_ijk, z_alg_f_x_ijk, ui, v_global, p);
+                            qj = s_sot*dcs.f_q_fun(x_ijk, z_ijk, ui, v_global, p);
+                            xk = opts.C_rk(1, kk+1) * x_lbp;
+                            for rr=1:opts.n_s
+                                x_ijr = obj.w.x(ii,jj,rr);
+                                xk = xk + opts.C_rk(rr+1, kk+1) * x_ijr;
+                            end
+                            obj.g.dynamics(ii,jj,kk) = {h * fj - xk};
+                            obj.g.z(ii,jj,kk) = {dcs.g_z_fun(x_ijk, z_ijk, ui, v_global, p)};
+                            obj.g.algebraic(ii,jj,kk) = {dcs.g_alg_fun(x_ijk, z_ijk, z_alg_ijk, v_global, p)};
+
+                            x_ij_end = x_ij_end + opts.D_rk(kk+1)*x_ijk;
+                            
+                            if opts.g_path_at_stg
+                                obj.g.path(ii,jj,kk) = {dcs.g_path_fun(x_ijk, z_ijk, ui, v_global, p), model.lbg_path, model.ubg_path};
+                            end
+                            if size(model.G_path, 1) > 0
+                                G_path = dcs.G_path_fun(x_ijk, z_ijk, ui, v_global, p);
+                                H_path = dcs.H_path_fun(x_ijk, z_ijk, ui, v_global, p);
+                                obj.G.path(ii,jj,kk) = {G_path};
+                                obj.H.path(ii,jj,kk) = {H_path};
+                            end
+                            if opts.cost_integration
```

Comment:
what happens if opts.cost_integration =  false? 
we have no objective?

Reply:
```matlab
if ~opts.cost_integration
    obj.f = obj.f + dcs.f_q_fun(x_ijk, z_ijk, u_i, v_global, p);
end
```

`cost_integration = false` yields what can be seen as a pointwise cost at the end of each control stage. This can be useful for e.g. MLE of parameters etc. 

We can rename but I think the last time we had this conversation we didn't have a better name.

Reply:
this is equivalent to a explicit Euler cost integration with a constant step size one. And we can name the option 
`Euler_cost_integration` ( hence `opts.Euler_cost_integration= true` is same as `opts.cost_integration = false`)` and explain in options what it does with a comment when to use it (MLE)

Reply:
Rename done and added more to documentation in `Options.m`